src/main/java/seedu/multitasky/logic/commands/ClearCommand.java

```diff
@@ -2,24 +2,90 @@
 
 import static java.util.Objects.requireNonNull;
 
+import seedu.multitasky.logic.parser.CliSyntax;
 import seedu.multitasky.model.EntryBook;
+import seedu.multitasky.model.entry.Entry;
 
+// @@author A0126623L
 /**
  * Clears the entry book.
  */
 public class ClearCommand extends Command {
 
     public static final String COMMAND_WORD = "clear";
-    public static final String MESSAGE_SUCCESS = "Entry book has been cleared!";
 
-    public static final String[] VALID_PREFIXES = {};
+    public static final String MESSAGE_USAGE = COMMAND_WORD + ": Clear entries in the entry book. " + "\n"
+                                               + "Format: " + COMMAND_WORD
+                                               + " [" + CliSyntax.PREFIX_ARCHIVE + " | "
+                                               + CliSyntax.PREFIX_BIN + " | "
+                                               + CliSyntax.PREFIX_ALL + "]" + "\n"
+                                               + "Example: " + COMMAND_WORD + " " + CliSyntax.PREFIX_BIN;
 
+    public static final String MESSAGE_ALL_SUCCESS = "Entry book has been cleared!";
+
+    public static final String MESSAGE_ACTIVE_SUCCESS = "Active entries have been cleared!";
+
+    public static final String MESSAGE_ARCHIVE_SUCCESS = "Archive has been cleared!";
+
+    public static final String MESSAGE_BIN_SUCCESS = "Bin has been cleared!";
+
+    public static final int NUMBER_ALLOWABLE_PREFIX = 1;
+
+    public static final String[] VALID_PREFIXES = { CliSyntax.PREFIX_ACTIVE.toString(),
+        CliSyntax.PREFIX_ARCHIVE.toString(),
+        CliSyntax.PREFIX_BIN.toString(),
+        CliSyntax.PREFIX_ALL.toString() };
+
+    public enum ClearType {
+        ACTIVE, ARCHIVE, BIN, ALL
+    }
+
+    private ClearType clearType;
+
+    public ClearCommand() {
+        clearType = ClearType.ACTIVE;
+    }
+
+    // @@author A0126623L
+    public ClearCommand(String prefix) {
+        /*
+         * TODO: Consider better ways than using multiple if-else?
```

Comment:
Not sure if switch-case is any more efficient or more compact than comparing strings though, since you'd first have to convert them to something like enum.